src/charm.py

```diff
@@ -42,12 +31,18 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
         self.framework.observe(self.on.minio_relation_joined, self.send_info)
```

Comment:
You'll also need to call `send_info` on `config-changed` as well.  I might also suggest moving the `is_leader` check up to the start of `__init__` since `send_info` is also going to be a no-op on non-leaders anyway.  (Just as a general pattern, I start all k8s charms' `__init__`s with an `is_leader` check and fast exit. :confused:)